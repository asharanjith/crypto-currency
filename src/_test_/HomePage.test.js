import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import HomePage from '../components/homePage/homePage';
import fetchCoinReducer from '../Redux/fetchCoin';

const store = configureStore({
  reducer: {
    allCoins: fetchCoinReducer,
  },
});

describe('HomePage', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <HomePage />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
