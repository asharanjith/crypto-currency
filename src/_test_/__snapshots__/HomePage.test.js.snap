// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render correctly 1`] = `
<div
  className="homePage"
>
  <form
    className="homePage__search"
  >
    <input
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
     
  </form>
  <div
    className="homePage__products"
  />
</div>
`;
